### ISSUE TITLE
CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource

### ISSUE THREAD

**Issue by meta-cast** [AzuraCast/AzuraCast#4900]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Stable Channel

### Current AzuraCast Version

0.14.1

### What happened?

Album art is not displaying because of the CORS error below (links are examples):

Access to XMLHttpRequest at 'https://example.com/api/station/1/art/12345.jpg' from origin 'https://cm.example.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by Abqehebter** [comment]
Small important question: You have already set the Access-Control-Allow-Origin header, right? (`Administration > Security & Privacy`)
![image](https://user-images.githubusercontent.com/42640247/146995521-52aab9ac-537e-421f-9ec2-cba556d7e9a1.png)
Works fine for me.

**Comment by meta-cast** [comment]
No, I was not aware of that option. Thank you. Do I have to restart docker?

**Comment by SlvrEagle23** [comment]
@meta-cast No, you shouldn't need to restart anything, but also that policy doesn't apply to images, as it's only recently that the concept of CORS included images at all (and I'm not even sure that _that_ is widespread, yet). I will have to investigate this and implement a solution into the system.

**Comment by abbadabbatech** [comment]
So how can the images be fixed, they are broken in Chrome and Edge, get same CORS error.

**Comment by SlvrEagle23** [comment]
Update: I misspoke in my previous message. After looking into it, it turns out that album artwork picks up the same Access-Control-Allow-Origin (ACAO) header as other API responses, so by setting it to `*` it should be fully permissive of any site including those images.

@abbadabbatech Have you tried updating that setting in your system settings? It should read simply `*`, not `"*"` or anything else similar.

**Comment by abbadabbatech** [comment]
The person that admins the system has made that change. Streams work fine
but images throw the CORS error.

**Comment by SlvrEagle23** [comment]
I'm unable to reproduce this error on the latest Rolling Release version, where every image I'm seeing from station artwork is also carrying with it the proper `Access-Control-Allow-Origin` header.

This may have been fixed in the time between the latest Stable release and the latest Rolling release.

**Comment by JMultimidia** [comment]
![image](https://user-images.githubusercontent.com/20319612/147980170-6f14cb88-e1dc-4299-bb17-0346414ddc30.png)

![image](https://user-images.githubusercontent.com/20319612/147980214-0a1904b4-0e8c-4d3b-9b66-0e5e343461c4.png)

Streams work fine but with CORS error.
…

**Comment by Abqehebter** [comment]
Does the URL `/live/` make a significant difference between them? I have set everything exactly as in your screenshots and it works without problems.

Try it with `"/api/nowplaying/radiotaim"`

**Comment by Vaalyn** [comment]
When I opened the URLs from the screenshots in my browser to see what headers the API returns I'm getting a `502 Bad Gateway` response in like 9/10 tries. Sometimes it actually does respond correctly.

I can see in the response of the `/api/nowplaying` endpoint that there are 42 stations on this server. This leads me to believe that the server is probably just overloaded and can't reliably deal with the load. Would be interesting to see what the nginx_proxy writes in its logs for the requests that are returning a `502 Bad Gateway`.
